src/main/java/agh/ics/oop/SimulationEngine.java

```diff
@@ -0,0 +1,31 @@
+package agh.ics.oop;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Vector;
+
+public class SimulationEngine implements IEngine {
+    private MoveDirection directions[];
+    private IWorldMap map;
+    private Vector2d initialCoordinates[];
+
+    public SimulationEngine(MoveDirection directions[], IWorldMap map, Vector2d initialCoordinates[]) {
+        this.directions = directions;
+        this.map = map;
+        this.initialCoordinates = initialCoordinates;
+        for(Vector2d initializingVector: initialCoordinates) {
+            map.place(new Animal(this.map, initializingVector));
+
+        }
+    }
+
+    public void run() {
+        for(int i = 0; i < directions.length; i++) {
+            //umyslnie korzystam z tego, ze animalToMove przechowuje referencje, aby ominac problem zmiany koordynatow, ale nie wiem, czy taka praktyka jest poprawna
+            Animal animalToMove = ((Animal)map.objectAt(initialCoordinates[i% initialCoordinates.length]));
```

Comment:
w zasadzie to ok, choć to mało wydajne rozwiązanie - za każdym razem musimy się dobrać do zwierzęcia po pozycji, co wymagana przejrzenia całej listy. Można by zamiast tego np. trzymać tu dodatkową listę zwierząt albo rozszerzyć itnerfejs o metodę `getAnimals()`